### ISSUE TITLE
StoragePreserveName not storing dirtree files under UID=dirtree

### ISSUE THREAD

**Issue by glbrimhall** [filesender/filesender#342]
Somewhere in the final push to get 2.0 released the StoragePreserveName got slightly broken when uploading a dirtree. 

When the client dirtree contains:

dir1
dir1/img3.jpg
dir1/img2.jpg
dir1/img1.jpg

What is seen on the filesystem server-side is:

./622a4a47-944e-5524-8de1-a732dc23d31d=dir1
./=dir1
./=dir1/img3.jpg
./=dir1/img2.jpg
./=dir1/img1.jpg

**Comment by monkeyiq** [comment]
hmm, I think the only updates IIRC post dirtree were internationalization, docs, and the improvement to TeraSender. 

Does the tree when dirtree was merged worked as expected? 
https://github.com/filesender/filesender/commit/a332e48a7b1e8a63f8b14bb3e0e038742646084d

**Comment by glbrimhall** [comment]
Ya, it did (at one point ;>)). Would be good to create an automated test
for this somehow.

-geoffb

**Comment by monkeyiq** [comment]
I think this is possible with the selenium stuff in the CI env. I'm tinkering trying to setup a local selenium server to make updating existing tests and creating new tests a quicker proposition.

**Comment by monkeyiq** [comment]
FWIW I managed to get selenium going locally. When I get this going on the CI for filesender I will make more info available on that process. This should help making new browser (and possible backend checks after) easier to do. I wasted a bunch of time trying to get firefox to work either natively or in docker.

This is from https://github.com/SeleniumHQ/docker-selenium
```
docker run -d -p 4444:4444 -p 5902:5900 -v /dev/shm:/dev/shm selenium/standalone-chrome-debug:3.12.0-cobalt
```

And I got a minor hack to this to test on that browser

```
vendor/bin/phpunit --verbose  vendor/sauce/sausage/WebDriverDemo.php
```

The above doesn't resolve this issue. Though as we both would like to add automated tests to stop such a regression this info helps along those lines.

**Comment by monkeyiq** [comment]
I assume that when the client dirtree contains:

dir1
dir1/img3.jpg
dir1/img2.jpg
dir1/img1.jpg

What is seen on the filesystem server-side is:

./622a4a47-944e-5524-8de1-a732dc23d31d=dir1
./=dir1
./=dir1/img3.jpg
./=dir1/img2.jpg
./=dir1/img1.jpg

What the expected filesystem server-side is:

./622a4a47-944e-5524-8de1-a732dc23d31d=dir1
./622a4a47-944e-5524-8de1-a732dc23d31d=dir1/img3.jpg
./622a4a47-944e-5524-8de1-a732dc23d31d=dir1/img2.jpg
./622a4a47-944e-5524-8de1-a732dc23d31d=dir1/img1.jpg

If this is the case then https://github.com/filesender/filesender/pull/365 changes to the expected for me, without rigorous testing.

**Comment by glbrimhall** [comment]
Thx ! I'm on vacation right now, will start playing with this when. I get
back.

-geoff Brimhall

**Comment by glbrimhall** [comment]
Fixed and submitted in pull request #370

**Comment by monkeyiq** [comment]
This was fixed with the merged https://github.com/filesender/filesender/pull/378
